lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -78,9 +80,16 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_metric('user_id', request.user.id)
         monitoring_utils.set_custom_metric('is_staff', request.user.is_staff)
 
-        course_tools = CourseToolsPluginManager.get_enabled_course_tools(request, course_key)
-
         course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+
+        _, request.user = setup_masquerade(
```

Comment:
Should this be done before the `course = get_course_with_access` above that uses the `request.user`?

Reply:
Great question. I copied this pattern (getting course first) from existing endpoints that used masquerading. :shrug: